src/main/java/games/strategy/engine/delegate/DelegateExecutionManager.java

```diff
@@ -33,7 +35,7 @@
    * only 1 block can be held (the block is equivalent to the read lock).
    */
   private final ReentrantReadWriteLock readWriteLock = new ReentrantReadWriteLock();
-  private final ThreadLocal<Boolean> currentThreadHasReadLock = ThreadLocal.withInitial(() -> Boolean.FALSE);
+  private final ThreadLocal<Boolean> currentThreadHasReadLock = ThreadLocal.withInitial(() -> false);
```

Comment:
Is there a specific reason the Boolean class objects get replaced with their according primitives that simply get casted back to instances of the Boolean class?

Reply:
No reason other than that we are typically not concerned with boxing/unboxing in the TripleA code (except maybe where performance is a concern).

Reply:
ff751b95a